src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexingService.cs

```diff
@@ -87,40 +114,50 @@ public async Task ProcessContentItemsAsync()
                         return;
                     }
 
-                    foreach (var task in batch)
-                    {
-                        var contentManager = scope.ServiceProvider.GetRequiredService<IContentManager>();
-                        var indexHandlers = scope.ServiceProvider.GetServices<IContentItemIndexHandler>();
+                    var contentManager = scope.ServiceProvider.GetRequiredService<IContentManager>();
+                    var indexHandlers = scope.ServiceProvider.GetServices<IContentItemIndexHandler>();
 
-                        foreach (var index in allIndices)
+                    foreach (var indexSettings in indexSettingsList)
+                    {
```

Comment:
We are still iterating on each indices I just moved it higher to be able to get the IndexSettings.LatestIndex for the contentManager query made at line 127.

Reply:
I understand that the logic is that the Lucene Background worker doesn't matter about which index to process it does them all.

Reply:
Good then, thanks